TERRAFORM_VALIDATOR.md

```diff
@@ -0,0 +1,65 @@
+# terraform-validator configuration
+
+These are instruction to configure [terraform-validator](https://github.com/GoogleCloudPlatform/terraform-validator) in CI/CD for the example foundation.
+
+These instructions consider that you have cloned `terraform-example-foundation` repo locally and had already configured the private repos for `gcp-org`, `gcp-environments`, `gcp-networks` and `gcp-projects`.
+
+## Run terraform-validator with tf-wrapper.sh script manually
+
+The user running terraform-validator must have at least these roles to run terraform-validator:
+
+- **Browser** - `roles/browser`
+- **Security Reviewer** - `roles/iam.securityReviewer`
+
+Installation steps:
+
+1. Install terraform-validator locally
+    1. Follow the instructions on the terraform-validator [user guide](https://github.com/forseti-security/policy-library/blob/master/docs/user_guide.md#how-to-use-terraform-validator) to download the latest version.
+    1. Rename `terraform-validator-linux-amd64` to `terraform-validator`
+    1. Update permissions `chmod +x terraform-validator`
+    1. Move terraform-validator binary to a folder in your PATH, like for example `$HOME/.local/bin`
+1. Outside of your clone of terraform-example-foundation repo clone the policy repo `git clone https://github.com/forseti-security/policy-library.git`
+1. Export the path to the policy repo `export policy_library_path=$(pwd)/policy-library`
+1. Change into one of you private repos like `gcp-org`
+1. Run terraform-validator
+    1. Run `./tf-wrapper.sh init prod`
+    1. Run `./tf-wrapper.sh plan prod`
+    1. Run `./tf-wrapper.sh validate  prod ${policy_library_path}`
+
+You should get a result like this:
+
+```
+*************** TERRAFORM VALIDATE ******************
+      At environment: envs/shared
+      Using policy from: /home/foo/policy-library
+*****************************************************
+ERROR: logging before flag.Parse: I0728 23:16:46.428406    3913 validator.go:177] starting 2 workers
+ERROR: logging before flag.Parse: I0728 23:16:46.428862    3913 validator.go:186] worker 0 starting
+ERROR: logging before flag.Parse: I0728 23:16:46.430272    3913 validator.go:186] worker 1 starting
+No violations found.
+```
+
+## Run terraform-validator in Cloud Build
+
+Configuration steps:
+
+1. The cloud build service account will need at least these roles to run terraform-validator:
+   - **Browser** - `roles/browser`
+   - **Security Reviewer** - `roles/iam.securityReviewer`
+1. Follow the instructions in [configure access for cloud build service account](https://cloud.google.com/cloud-build/docs/securing-builds/configure-access-for-cloud-build-service-account) to grant these roles.
```

Comment:
Why do they need to do this manually? These should be baked into the bootstrap.

Reply:
+1